src/librepgp/stream-write.cpp

```diff
@@ -98,6 +102,28 @@ typedef struct pgp_dest_encrypted_param_t {
     size_t                     chunkidx; /* index of the current AEAD chunk */
     size_t                     cachelen; /* how many bytes are in cache, for AEAD */
     uint8_t cache[PGP_AEAD_CACHE_LEN];   /* pre-allocated cache for encryption */
+#ifdef ENABLE_CRYPTO_REFRESH
+    std::array<uint8_t, PGP_SEIPDV2_SALT_LEN> v2_seipd_salt; /* SEIPDv2 salt value */
+#endif
+
+    bool
+    is_aead_auth()
+    {
+        return (this->auth_type == rnp::AuthType::AEADv1)
```

Comment:
Agreed. Since `auth_type` is an enum class, the compiler will generate a warning if not all cases are handled. This is an advantage if other authentication types should be added later. Will be addressed in upcoming commit.